Enterprise-CMCS/macpro-mdct-mfp
# [CMDCT-159]: SAR Form Creation Cleanup Part 1

### Description


- Abstracts out code during the creation process in the api handler
- Adds a TON of comments to help the reader follow along
- Updates the value sent for the finalSar question to be in the proper Choice type format

To best read this PR, I recommend you pull it down and read the create.ts file from top to bottom. That'll be much easier to understand and review than if you try and use the Files Changed mode. I had to move some code around to make it read better so you might lose track of things or wonder why I'm deleting them when really its added back later on or further up. You can also try to read it by clicking through each commit. I tried to package everything so that the commit message makes sense for what I changed. I recommend doing that after reading create.ts separately.

### Related ticket(s)


CMDCT-159

---
### How to test


- Create a work plan
- Create a SAR
- See that you are unable to create an additional SAR (Assuming you only created the one WP)
- Check the network tab and refresh the page on the dashboard page for SAR and Workplan
- Note that there should be a respective field called sar or workplan in the response with the id of the SAR or WP its connected to
- Create 2 work plans
- Edit a field in both work plans but make those values different (So you can tell the work plans apart)
- Create a SAR and see that it built it of the third work plan you just created, not the second or first one.
- Create a SAR and see that it was built off the second work plan you created.
- Try editing a SAR, note that your answers should all be correct
- See that you can't edit a WP too

---
### Author checklist
